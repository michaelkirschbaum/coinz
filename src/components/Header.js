/** @jsx jsx */
import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import { css, jsx } from '@emotion/core'
import { UserContext } from '../App'
// import logo from '../../public/logo.jpg'

const Header = ({ setUser }) => {
  const user = useContext(UserContext)

  return (
    <div css={css`
      display: flex;
      justify-content: space-between;
      height: 40px;
      line-height: 40px;
      border-bottom: 1px solid lightgrey;
    `}>
      <div>
        <img alt='logo' />
        {user}
      </div>
      <div>
        Signed In
        <button onClick={() => setUser('')}>Sign Out</button>
      </div>
    </div>
  )
}

Header.propTypes = {
  setUser: PropTypes.func.isRequired,
}

export default Header