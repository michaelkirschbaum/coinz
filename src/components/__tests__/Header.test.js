import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Header'

describe('Header', () => {
  it('should match the snapshot', () => {
    const header = shallow(
      <Header />,
    )
    expect(header).toMatchSnapshot()
  })

  it('should render account address', () => {
    expect(false).toBe(true)
  })
})