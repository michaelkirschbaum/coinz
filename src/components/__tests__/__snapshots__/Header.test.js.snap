// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Header"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "map": undefined,
      "name": "5n91x2",
      "next": undefined,
      "styles": "
      float: right; 
      margin-right: 25px;
    ",
    }
  }
>
  <img
    alt="logo"
  />
  <div>
    Alice
     Signed In
  </div>
  <button
    onClick={[Function]}
  >
    Sign Out
  </button>
</EmotionCssPropInternal>
`;
